x-pack/plugins/security_solution/public/cloud_security_posture/components/misconfiguration/misconfiguration_overview.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { css } from '@emotion/react';
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText, useEuiTheme, EuiTitle } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { DistributionBar } from '@kbn/security-solution-distribution-bar';
+import { useMisconfigurationPreview } from '@kbn/cloud-security-posture/src/hooks/use_misconfiguration_preview';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { i18n } from '@kbn/i18n';
+import { ExpandablePanel } from '@kbn/security-solution-common';
+import { buildMisconfigurationPreviewQuery } from '@kbn/cloud-security-posture-common';
+
+export const MisconfigurationsOverview = ({ hostName }: { hostName: string }) => {
+  const hostNameQuery = useMemo(() => {
+    return buildMisconfigurationPreviewQuery('host.name', hostName);
+  }, [hostName]);
```

Comment:
Using `useMemo` here is unnecessary and could lead to increased memory usage, as the `buildMisconfigurationPreviewQuery` function is not computationally expensive.